t/50mruby-task.t

```diff
@@ -0,0 +1,123 @@
+                or die "failed to extract port number";
+            check_port($1);
+        },
+    );
+};
+
+subtest "single task with multiple http requests" => sub {
+    my $server = spawn_h2o(<< 'EOT');
+hosts:
+  default:
+    paths:
+      /:
+        mruby.handler: |
+          Proc.new do |env|
+            ch = create_channel
+            req_url = "http://$upstream_hostport/index.txt"
+            req1 = http_request(req_url)
+            req2 = http_request(req_url)
+            req3 = http_request(req_url)
+            task { req1.join; ch.push "1"; req2.join; ch.push "3"; req2.join; ch.push "5"; }
+            res = ""
+            res += ch.shift
+            res += "2"
+            res += ch.shift
+            res += "4"
+            res += ch.shift
+            res += "6"
+            [200, {}, [res]]
+          end
+
+EOT
+    my ($stderr, $stdout) = run_prog("curl --silent --dump-header /dev/stderr http://127.0.0.1:$server->{port}/");
+    is $stdout, "123456";
+};
+
+subtest "multiple tasks with multiple http requests" => sub {
+    my $server = spawn_h2o(<< 'EOT');
+hosts:
+  default:
+    paths:
+      /:
+        mruby.handler: |
+          Proc.new do |env|
+            ch = create_channel
+            req_url = "http://$upstream_hostport/index.txt"
+            req1 = http_request(req_url)
+            req2 = http_request(req_url)
+            req3 = http_request(req_url)
+            task { req1.join; ch.push "1" }
+            task { req2.join; ch.push "1"  }
+            task { req3.join; ch.push "1"  }
+            res = ""
+            res += ch.shift
+            res += "2"
+            res += ch.shift
+            res += "2"
+            res += ch.shift
+            res += "2"
+            [200, {}, [res]]
+          end
+
+EOT
+    my ($stderr, $stdout) = run_prog("curl --silent --dump-header /dev/stderr http://127.0.0.1:$server->{port}/");
+    is $stdout, "121212";
+};
+
+subtest "multiple tasks with multiple channel" => sub {
+    my $server = spawn_h2o(<< 'EOT');
+hosts:
+  default:
+    paths:
+      /:
+        mruby.handler: |
+          Proc.new do |env|
+            ch1 = create_channel
+            ch2 = create_channel
+            ch3 = create_channel
+            req_url = "http://$upstream_hostport/index.txt"
+            req1 = http_request(req_url)
+            req2 = http_request(req_url)
+            req3 = http_request(req_url)
+            task { req1.join; ch1.push "1" }
+            task { req2.join; ch2.push "3"  }
+            task { req3.join; ch3.push "5"  }
+            res = ""
+            res += ch1.shift
+            res += "2"
+            res += ch2.shift
+            res += "4"
+            res += ch3.shift
+            res += "6"
+            [200, {}, [res]]
+          end
+
+EOT
+    my ($stderr, $stdout) = run_prog("curl --silent --dump-header /dev/stderr http://127.0.0.1:$server->{port}/");
+    is $stdout, "123456";
+};
+
+done_testing();
```

Comment:
I would appreciate if you could add the tests which joins response body :D

Reply:
I added a test for that.
https://github.com/h2o/h2o/pull/1336/commits/250acdd7c8e59b84b3460db4b6c1d7ff9bf63842